packages/aws-cdk-lib/custom-resources/test/custom-resource-config/custom-resource-config.test.ts

```diff
@@ -0,0 +1,158 @@
+import { Capture, Template, Match } from '../../../assertions';
+import * as logs from '../../../aws-logs';
+import * as s3 from '../../../aws-s3';
+import * as s3deploy from '../../../aws-s3-deployment';
+import * as cdk from '../../../core';
+import { CustomResourceConfig } from '../../lib/custom-resource-config/custom-resource-config';
+
+describe('when logging is undefined', () => {
+
+  test('when CustomResourceConfig.addLogRetentionLifetime is called, singleton-backed custom resource has new logGroup with retention period specified', () => {
+    const setLogRetention = logs.RetentionDays.TEN_YEARS;
+    const app = new cdk.App();
+    CustomResourceConfig.of(app).addLogRetentionLifetime(setLogRetention);
+    const stack = new cdk.Stack(app);
+
+    let websiteBucket = new s3.Bucket(stack, 'WebsiteBucket', {});
+    new s3deploy.BucketDeployment(stack, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+    });
+
+    const template = Template.fromStack(stack);
+
+    template.hasResourceProperties('AWS::Lambda::Function', {
+      LoggingConfig: {
+        LogGroup: {
+          Ref: Match.stringLikeRegexp('CustomCDKBucketDeployment'),
+        },
+      },
+    });
+    template.resourceCountIs('AWS::Logs::LogGroup', 1);
+    template.hasResourceProperties('AWS::Logs::LogGroup', {
+      RetentionInDays: setLogRetention,
```

Comment:
```suggestion
      RetentionInDays: logRetention,
```

you don't need to prefix the variables with `set`